epoxy-adapter/src/main/java/com/airbnb/epoxy/EpoxyModelWithView.java

```diff
@@ -0,0 +1,58 @@
+package com.airbnb.epoxy;
+
+import android.support.annotation.LayoutRes;
+import android.view.View;
+import android.view.ViewGroup;
+
+/**
+ * A model that allows its view to be built programmatically instead of by inflating a layout
+ * resource. Just implement {@link #buildView} so the adapter can create a new view for this model
+ * when needed.
+ * <p>
+ * {@link #getViewType()} is used by the adapter to know how to reuse views for this model. If it is
+ * left unimplemented then the generated model will include an implementation which returns a value
+ * based on the generated model's name. This means that all models of that type should be able to
+ * share the same view, but the view won't be shared with models of any other type.
+ * <p>
+ * The generated view type will be negative so that it cannot collide with values from layout
+ * resources, which are used in normal Epoxy models. However, it is possible for generated view
+ * types to collide if models from different libraries are used together, since the model processor
+ * can only guarantee that view types are unique within a library. If you need to use {@link
+ * EpoxyModelWithView} models from different libraries or modules together you can define a view
+ * type manually to avoid this small collision chance. A good way to manually create value is by
+ * creating an R.id. value in an ids resource while, which will guarantee a unique value.
```

Comment:
```
class ViewTypeManager {
  private final Map<Class<? extends EpoxyModel>, int> viewTypeMap = new HashMap<>();

  public int getViewType(EpoxyModel model) {
    Class<? extends EpoxyModel> modelClass = model.getClass();
    if (!viewTypeMap.contains(modelClass)) {
      int viewType = -viewTypeMap.size()-1;
      viewTypeMap.put(modelClass, viewType);
      return viewType;
    } else {
      return viewTypeMap.get(modelClass);
    }
  }
}
```

How about using a singleton to generate unique View types for every model class?